build/LocalizationTasks/SaveXlfToResx.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using xlflib;
+
+namespace Microsoft.Build.LocalizationTasks
+{
+    public class SaveXlfToResx : Task
+    {
+        [Required]
+        public string ResxPath { get; set; }
+
+        [Required]
+        public string XlfPath { get; set; }
+
+        /// <summary>
+        /// When set to "true" it makes the task replace the resource values with another value.
+        /// Useful for testing localized builds.
+        /// 
+        /// If <see cref="TestString"/> is non empty, resource values get replaced with it.
+        /// If it is empty, resource values get replaced with !resource_id!english_resource!localized_resource!
+        /// </summary>
+        public string ReplaceWithTestString { get; set; }
+
+        public string TestString { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                var xlfDocument = new XlfDocument(XlfPath);
+
+                if (!string.IsNullOrEmpty(ReplaceWithTestString) && ReplaceWithTestString.Equals("true"))
+                {
+                    ReplaceResourceValuesWithTestString(xlfDocument);
+                }
+
+                xlfDocument.SaveAsResX(ResxPath);
+
+                return true;
+            }
+            catch (Exception e)
```

Comment:
The standard advice, which I'm mostly just parroting here, is that catching `Exception` is too general, because it catches 
* Interesting runtime failures that might be actionable (what you want)
* Runtime failures due to programmer error that are bugs but not actionable for the end user (_maybe_ ok, might be better represented elsewhere)
* Systemic failures due to process or machine state (when you mostly just want to crash)

MSBuild itself already [provides an all-purpose task-exception handler](https://github.com/Microsoft/msbuild/blob/4b30fac51ab58dadfffedb2a57e83b8398e3a5e1/src/XMakeBuildEngine/BackEnd/Components/RequestBuilder/TaskBuilder.cs#L773-L902), so I'd say to just depend on that if there's no `XliffParserException` that's specific to the code that's being called. Plus, it doesn't look like there's [much in the way of exceptions](https://github.com/fmuecke/XliffParser/search?utf8=%E2%9C%93&q=throw) in there, currently.

Short further reading: https://msdn.microsoft.com/en-us/library/ms182137.aspx
LONG but interesting further reading: [Midori error model](http://joeduffyblog.com/2016/02/07/the-error-model/)